### PR TITLE
fix: prevent cursor-overflow panic on PR reload that shrinks the diff

### LINKED ISSUE #573: panic when reloading a PR of shorter length
I was using tuicr local built from this branch. I loaded up a PR while I had the main branch checked out. In another bash window, I checked out the PRs source branch. Made a change. Pushed the change remote. When I swithched back to the other bash window running tuicr, I issued command :e to refresh the pr, and tuicr panic crashed w error; 

[11:00]  Bash main ≡                                                                                                                                                            1s 391ms
   ~\source\repos\Comp.Submissions
  !❯ ../tuicr/target/debug/tuicr.exe
  tuicr-session: _git/Comp.Submissions@main/worktree/06ca818

  thread 'main' (13296) panicked at src\app\navigation.rs:54:32:
  attempt to subtract with overflow
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

### PR DESCRIPTION
**Pull request by joshvito**
fixes #573

A PR `:e` reload can replace the diff with a shorter one while leaving `cursor_line` parked past the new end (the same-head reload branch, and a restored overview cursor captured from the taller old diff, never reclamp it). The next `cursor_down` then clamps the cursor *up* to the new `max_cursor_line` — below `prev_cursor` — underflowing the `cursor_line - prev_cursor` usize subtraction and panicking with "attempt to subtract with overflow" at navigation.rs.

- cursor_down: use `saturating_sub` for the cursor-movement delta, matching the existing `cursor_up` siblings.
- finish_pr_reload / reload_pull_request_with_backend: clamp `cursor_line` into the reloaded diff's bounds after the swap/restore.
- add a regression test that reproduces the stale-cursor overflow.

### CODE REVIEW

**Comment by joshvito** [comment]
I want to test this before I publish.

File: src/app/tests/scroll_behavior_tests.rs [hunk 1]
```diff
@@ -725,3 +725,27 @@ fn move_cursor_to_annotation_wrap_on_scrolls_target_fully_into_view() {
         "already-visible target should not change scroll_offset"
     );
 }
+
+#[test]
+fn cursor_down_with_stale_cursor_past_end_does_not_underflow() {
+    // Regression: a PR reload can shrink the diff and leave `cursor_line`
```
Comment (joshvito) [inline src/app/tests/scroll_behavior_tests.rs]: just testing that this change didn't break basic commenting functionality;

**Review by agavra — APPROVED** [review]
nice catch @joshvito 🙏
